torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -728,11 +735,21 @@ Tensor prod_backward(
     return grad;
   }
   dim = at::maybe_wrap_dim(dim, input.sizes().size());
-  if (!keepdim && input.dim() != 1) {
+  if (!keepdim) {
+    // `prod` reduces the dimension at `dim`,
+    // so, unsqueeze `grad` and `result` at dim.
     grad = grad.unsqueeze(dim);
     result = result.unsqueeze(dim);
   }
-  if (is_meta_in_composite_kernels(input)) {
+  if (input.numel() == 0) {
+    // When input has a zero sized dimension (empty tensor),
+    // we don't need to actually compute the grads.
+    // So we just reshape `grad` as `input`.
+    return grad.expand_as(input);
+  }
```

Comment:
Can we put this inside `prod_safe_zeros_backward` and deduplicate with the other call site above?

Reply:
Yup. Will do that. Thanks :)